src/types/fetch.ts

```diff
@@ -51,9 +51,13 @@ export type AvailableRouterMethod<R extends NitroFetchRequest> =
 
 // Argumented fetch options to include the correct request methods.
 // This overrides the default, which is only narrowed to a string.
-export interface NitroFetchOptions<R extends NitroFetchRequest>
-  extends FetchOptions {
-  method?: Uppercase<AvailableRouterMethod<R>> | AvailableRouterMethod<R>;
+export interface NitroFetchOptions<
+  R extends NitroFetchRequest,
+  M extends AvailableRouterMethod<R> = AvailableRouterMethod<R>
+> extends FetchOptions {
+  method?:
+    | Uppercase<AvailableRouterMethod<R> & M>
+    | (AvailableRouterMethod<R> & M);
```

Comment:
Is there a reason this couldn't be:

```suggestion
  method?: Uppercase<M> | M;
```

Reply:
No, not really. Fixed